java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -466,44 +398,30 @@ public EnumConstantTreeImpl newEnumConstant(
     Optional<ClassTreeImpl> classBody,
     Optional<InternalSyntaxToken> semicolonToken) {
 
+    // FIXME SONARJAVA-547 Handle annotations
     IdentifierTreeImpl identifier = new IdentifierTreeImpl((InternalSyntaxToken) identifierToken);
-    if (annotations.isPresent()) {
-      identifier.prependChildren(annotations.get());
-    }
 
-    List<JavaTree> children = Lists.newArrayList();
     SyntaxToken openParenToken = null;
     SyntaxToken closeParenToken = null;
-    List argumentsList = Collections.emptyList();
+    List<ExpressionTree> argumentsList = Collections.emptyList();
     if (arguments.isPresent()) {
       ArgumentListTreeImpl argumentsListTreeImpl = arguments.get();
       argumentsList = argumentsListTreeImpl;
       openParenToken = argumentsListTreeImpl.openParenToken();
       closeParenToken = argumentsListTreeImpl.closeParenToken();
-      children.add(argumentsListTreeImpl);
-    }
-
-    if (classBody.isPresent()) {
-      children.add(classBody.get());
     }
 
     NewClassTreeImpl newClass = new NewClassTreeImpl(
       openParenToken,
       argumentsList,
       closeParenToken,
-      classBody.isPresent() ? classBody.get() : null,
-      children.toArray(new AstNode[0]));
+      classBody.isPresent() ? classBody.get() : null
+      );
     newClass.completeWithIdentifier(identifier);
 
     @SuppressWarnings("unchecked")
     EnumConstantTreeImpl result = new EnumConstantTreeImpl(modifiers((Optional<List<ModifierTree>>) (Optional<?>) annotations), identifier, newClass);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)